### PR TITLE
add: Whisper perks (White Nail/Whispered Breathing/crafting)

### PR DESCRIPTION
**Pull request by FlaminSarge**

### CODE REVIEW

File: src/perks/exotic_perks.rs [hunk 1]
```diff
@@ -1247,5 +1247,33 @@ pub fn exotic_perks() {
...
+                refund_mag: 3,
+                refund_reserves: -2,
+            }
+        }),
+    );
+
+    add_dmr(
+        Perks::WhisperedBreathing,
+        Box::new(|_input: ModifierResponseInput| -> DamageModifierResponse {
+            if _input.value > 0 {
```
Comment (FlaminSarge) [inline src/perks/exotic_perks.rs:1267]: Unsure if I should be trying something with calc_data or firing_data here, we don't have crits_landed or anything to disable the perk if hits landed > crits landed.

File: src/perks/exotic_perks.rs [hunk 2]
```diff
@@ -1247,5 +1247,33 @@ pub fn exotic_perks() {
...
+                let mut crit_mult =
+                    (_input.calc_data.base_crit_mult + 1.0) / _input.calc_data.base_crit_mult;
```
Comment (FlaminSarge) [inline src/perks/exotic_perks.rs:1269]: Will doublecheck this value before moving out of draft

File: src/perks/exotic_perks.rs (hunk 1 above)
Comment (FlaminSarge) [reply src/perks/exotic_perks.rs:1267]: nvm the perk stays even if you miss a crit, it's the lack of refund forcing you to unscope that un-procs it. we're good here, I think.

**Review by BarmonHammer — APPROVED** [review]
good to merge, we will need to come back and make sure the values are still correct since TFS sniper update

**Pull request merged**